### ISSUE TITLE
Native Apple ARM64 Support

### ISSUE THREAD

**Issue by Diatrus** [input-output-hk/daedalus#2677]
Creating issue in hopes of having an installer for a native M1 version of Daedalus

**Comment by miorsufianiohk** [comment]
Dear @Diatrus ,

Thank you for the request. We do have a plan to release the arm version of Daedalus. Once, it's completed we will let everyone know via our normal communication channel.

Kind regards,
Mior

**Comment by miorsufianiohk** [comment]
Hi @danielmain, 

Created Jira ticket is [here ](https://input-output.atlassian.net/browse/DDW-745)

Thanks,
Mior

**Comment by mark-stopka** [comment]
https://github.com/input-output-hk/daedalus/pull/2684

**Comment by chisNaN** [comment]
Hello Thanks for this adaptation.
I am also really interested in the arm version and prefer to wait instead of installing a rosetta emulation...
Any estimation ?
Thanks in advance

**Comment by danielmain** [comment]
Hi @chisNaN we are very close to get a Daedalus aarch64 (M1) version.

If you need it now you can build it from the sources using the branch `feature/ddw-745-arm-support`. Or check our [PR #2684](https://github.com/input-output-hk/daedalus/pull/2684)

Or wait until we integrate the compilation in our CI process ;)

**Comment by chisNaN** [comment]
Thanks for your reactivity and instructions @danielmain but if "you are very close" indeed i prefer to wait.
I am with cardano and daedalus since end 2017 and hope to follow this ambitious project longer

**Comment by danielmain** [comment]
Hi @chisNaN "very close" is something relatively 😅 but from what I can say, using the source code you can have a native M1 Daedalus version today.

Correct me pls @disassembler we need to integrate real M1 macs on our CI process before we can produce a native aarch64-Daedalus. Samuel, do you know when is that going to happen? Thank you!

**Comment by chisNaN** [comment]
Hello folks.
Any update for the arm with the recent 4.8 release?
Have a good week

**Comment by danielmain** [comment]
We are waiting this PR to be merged first https://github.com/input-output-hk/cardano-wallet/pull/2767

**Comment by chisNaN** [comment]
Hello. I saw daedalus flight is on 4.9.
Any news for the arm build one?
Thanks

**Comment by danielmain** [comment]
Hi @chisNaN

Seems simple, but there is a lot of work under the hood 😵

1. The mentioned PR needs to be merged, then you can build from the sources a M1-native Daedalus.
2. Additionally, we are upgrading our CI infrastructure to build/sign/ship the new Daedalus so that you can soon download it from [daedaluswallet.io](https://daedaluswallet.io/) in case you prefer not to build it by yourself.
3. Furthermore, our QA Team needs to do an extensive regression test on the new native build before giving it to the public.

**Comment by chisNaN** [comment]
Thanks for your reply
Honestly as i have managed decent amount of ADA i prefer to wait for a simple to db click on dmg
(even if i am full stack web dev)

**Comment by chisNaN** [comment]
From what i can read here https://github.com/input-output-hk/cardano-wallet/pull/2767

it's seems almost available

**Comment by chisNaN** [comment]
Hello folks 
Any updates on this one?
thanks in advance

**Comment by danielmain** [comment]
Hi @chisNaN we have a running branch already, you can run it today. It can be run on your machine if you compile it. 

An Installable artifact is not yet ready since our DevOps team needs to set up and install M1 machines in our CI infrastructure. cc: @disassembler 

If you follow the [README.md](https://github.com/input-output-hk/daedalus/tree/feature/ddw-745-arm-support#linuxmacos) you can have it now or wait for an official release. Thank you for your patience, but it is not as simple as it seems 😞.

<img width="1240" alt="image" src="https://user-images.githubusercontent.com/4303099/167582728-1e57421a-794a-4d9a-a272-c3c5aa15804d.png">

**Comment by chisNaN** [comment]
THanks a lot will do that

**Comment by danielmain** [comment]
@chisNaN if you check the branch `feature/ddw-745-arm-support` that is in this [PR](https://github.com/input-output-hk/daedalus/pull/2684) you will see that as soon it gets merged in develop branch it will look like this: 
<img width="874" alt="grafik" src="https://user-images.githubusercontent.com/4303099/169783698-0a485917-a84c-4472-9fff-e42468461788.png">

**Comment by chisNaN** [comment]
THANKS a lot will try that then

**Comment by simonmichael** [comment]
Great, I appreciate the work on this, many of us are keenly ready to waste less cpu power and speed up wallet catchup. 

Why "close the issue as completed" ? That's confusing.

**Comment by danielmain** [comment]
Hi @simonmichael builds for aarch64-darwin (apple M1) can already be downloaded at https://buildkite.com/input-output-hk/daedalus/builds/22165#0182f32d-45de-420b-97c8-3df67fc992a7

**Just, those builds are not yet signed with the IOG certificate**, so you will see a warning by installing.

Our colleague @disassembler has to setup apple-M1 machines in our CI first so we can sign the installers. After that you will see a new download option in [www.daedaluswallet.io/download](https://daedaluswallet.io/download/)

**Comment by chisNaN** [comment]
Hello danielmain how are you and thanks
Is it this link in the "Artifact" section?

https://buildkite.com/organizations/input-output-hk/pipelines/daedalus/builds/22165/jobs/0182f32d-45de-420b-97c8-3df67fc992a7/artifacts/0182f342-d972-4bbf-9ae9-94bc2d73f749

**Comment by danielmain** [comment]
Hi @chisNaN this native M1 build corresponds to Daedalus 5.1.1 (latest version today) https://buildkite.com/input-output-hk/daedalus/builds/22358#01840989-ff30-46ad-b983-8bbac3a105a7

And yes, go to artifacts and download the `[UNSAFE-internal-build/daedalus-5.1.1-mainnet-22358-aarch64-darwin.pkg](https://buildkite.com/organizations/input-output-hk/pipelines/daedalus/builds/22358/jobs/01840989-ff30-46ad-b983-8bbac3a105a7/artifacts/01840995-2541-4878-86df-eac483bfeff1)`

It is called UNSAFE because it has not yet been signed with the IOG certificate. You will probably get a warning by installing it. We are still working to add M1/M2 apple machines in our data center to automate the signing process. 

![image](https://user-images.githubusercontent.com/4303099/204096479-234b711d-5482-4c9c-9c5b-3a5ee0a7e391.png)

**Comment by chisNaN** [comment]
@danielmain thanks a lot

**Comment by simonmichael** [comment]
Checking in. Thanks as always for the work on this. I guess running unsigned binaries, even if rather official looking, even (especially) if one is technically confident, might be the wrong direction. Looking forward to an official release.

**Comment by simonmichael** [comment]
PS, if there's ways non-IOG folk can help, let us know.

**Comment by danielmain** [comment]
Hi @simonmichael, only IOG owns the keys to sign the binaries. @disassembler has it in the TODO list. From my side, I cannot do more than offer you these unsigned binaries.

But if security is your concern, **you can build Daedalus from the source!** That is even more secure than trusting someone else to sign it.

**Comment by chisNaN** [comment]
Hello Folks happy new year.
Any information about official apple silicon release?
thanks

**Comment by danielmain** [comment]
https://github.com/input-output-hk/cicero/issues/78

**Comment by simonmichael** [comment]
Pinging again as https://github.com/input-output-hk/cicero/issues/78 seems complete.

**Comment by sebasmannem** [comment]
It would be awesome if we could have an AARCH64 version available through official channels

**Comment by terbed** [comment]
Is there an update on this? Are there plans to provide Apple silicone support through official channels? Or will it not be supported by default (and we can close this issue)?

**Comment by disassembler** [comment]
Closing as part of issue backlog cleanup — issues older than 90 days are being retired to give the new team a clean slate. Please open a new issue if this is still relevant.
